### ISSUE TITLE
CCA/CKA/OPD support

### ISSUE THREAD

**Issue by brando90** [meta-pytorch/captum#572]
Is CCA or CKA or Orthogonal Procrustes Distance (OPD) available for captum?

**Comment by brando90** [comment]
https://discuss.pytorch.org/t/cca-cka-for-captum/107084

**Comment by brando90** [comment]
https://discuss.pytorch.org/t/what-is-a-good-cca-cka-library-for-pytorch-that-works-ideally-with-gpu/104889

**Comment by NarineK** [comment]
@brando90, thank you for the question. No, we don't have CCA / CKA implementation in captum library. We thought about it but we haven't added it yet.

**Comment by brando90** [comment]
> @brando90, thank you for the question. No, we don't have CCA / CKA implementation in captum library. We thought about it but we haven't added it yet.

Thanks fo your reply Nerine!

What do you think of merging incorporating [anatome's CCA/CKA](https://github.com/moskomule/anatome) etc into captum?

**Comment by NarineK** [comment]
Thank you @brando90  for sharing the link. I'll look deeper into it and estimate the efforts required to implement it. You mentioned that currently anatome doesn't work reliably and ideally you'd like to have a more reliable implementation of the algorithms implemented there. Is that right ?

**Comment by pietrolesci** [comment]
Hi @brando90,

Thanks for your answer. I definitely think that this can be a way forward. Though, having everything in Captum was a quite captivating idea (pun intended). I will definitely use your fork for now and will monitor Captum development to see whether this will be supported.

Best,
Pietro

**Comment by brando90** [comment]
> Though, having everything in Captum was a quite captivating idea (pun intended).

Agreed. How else do you recommend going forward? I think the best is:

1. reproduce as many tests from google's SVCCA tutorial (to gain confidence)
2. then once we know we have a slightly solid version of my ultimate-anatome, push it to Captum - to captive future users ;) 

Let me know if you think there is a better alternative. :) 

Regards, Brando

**Comment by NarineK** [comment]
We have done some work CCA/CKA that hasn't been merged to master yet. @bilalsal can tell more about it.

**Comment by pietrolesci** [comment]
Hi @NarineK,

Thanks a lot for your answer. 

To add to the discussion, I have recently [found out](https://twitter.com/AnandkrishAK/status/1455120040194314249) that @AntixK (CC'ed) has published the [Pytorch-Model-Compare](https://github.com/AntixK/PyTorch-Model-Compare) library. I think there can be value in coordinating the efforts on this topic - I hope this can be useful to centralize the discussion.

Best,
Pietro

**Comment by brando90** [comment]
thanks!

Not sure how we can do that, perhaps pinging them here might be best, but for now my ultimate-anatome is working and I've put enough effort into it that that's what I will stick with for now unless there are excellent reasons to switch. I agree it's a pitty for the distributed efforts, alas! I guess it happens when people are secretive about research or perhaps it's hard to google with slightly different names.

Thanks Pietro.

**Comment by joshua-xia** [comment]
any progress? It is very important feature, Thanks!
